File: qc-core/src/main/java/com/eager/questioncloud/core/domain/verification/infrastructure/EmailVerificationJpaRepository.java
package com.eager.questioncloud.core.domain.verification.infrastructure;

import com.eager.questioncloud.core.domain.verification.enums.EmailVerificationType;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface EmailVerificationJpaRepository extends JpaRepository<EmailVerificationEntity, String> {
    Optional<EmailVerificationEntity> findByTokenAndEmailVerificationTypeAndIsVerifiedFalse(
        String token, EmailVerificationType emailVerificationType);

    Optional<EmailVerificationEntity> findByResendToken(String resendToken);
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/verification/infrastructure/repository/EmailVerificationJpaRepository.kt
package com.eager.questioncloud.core.domain.verification.infrastructure.repository

import com.eager.questioncloud.core.domain.verification.enums.EmailVerificationType
import com.eager.questioncloud.core.domain.verification.infrastructure.entity.EmailVerificationEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface EmailVerificationJpaRepository : JpaRepository<EmailVerificationEntity, String> {
    fun findByTokenAndEmailVerificationTypeAndIsVerifiedFalse(
        token: String,
        emailVerificationType: EmailVerificationType
    ): Optional<EmailVerificationEntity>

    fun findByResendToken(resendToken: String): Optional<EmailVerificationEntity>
}
